docs/topics/advent-of-code.md

```diff
@@ -0,0 +1,74 @@
+when each puzzle is released daily. With the kind permission of [Eric Wastl](http://was.tl/), creator of Advent of Code,
+we demonstrate how to solve these puzzles using the idiomatic Kotlin style.
+
+> You can find all solutions for Advent of Code 2020 puzzles in our [GitHub repository](https://github.com/kotlin-hands-on/advent-of-code-2020/).
+>
+
+## Day 1. Report repair
+
+In this puzzle, you'll explore input handling, iteration over the list, different ways of building a map, and use
+the `let` function to simplify your code.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/1)
+* Check out the solution by Svetlana Isakova in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/07/advent-of-code-in-idiomatic-kotlin/)
+or watch the video:
+
+<video width="560" height="315" href="o4emra1xm88" title="Kotlin Tutorial: Advent of Code Puzzles, Day 1"/>
+
+## Day 2. Password philosophy
+
+In this puzzle, you'll explore string utility functions, regular expressions, operations on collections,
+and how the `let` function can be helpful to transform your expressions.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/2)
+* Check out the solution by Svetlana Isakova in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/07/advent-of-code-in-idiomatic-kotlin-day2/)
+or watch the video:
+
+<video width="560" height="315" href="MyvJ7G6aErQ" title="Kotlin Tutorial: Advent of Code Puzzles, Day 2"/>
+
+## Day 3. Toboggan trajectory
+
+In this puzzle, you'll compare imperative and more functional code styles, work with pairs and the `reduce` function,
+edit code in column selection mode, and fix integer overflows.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/3)
+* Check out the solution by Mikhail Dvorkin on [GitHub](https://github.com/kotlin-hands-on/advent-of-code-2020/blob/master/src/day03/day3.kt)
+or watch the video:
+
+<video width="560" height="315" href="ounCIclwOAw" title="Kotlin Tutorial: Adopting a Functional Style for Advent of Code Puzzles"/>
+
+## Day 4. Passport processing
+
+In this puzzle, you'll apply the `when` statement and explore different ways of how to validate the input — using
+utility functions, working ranges, checking set membership, and matching a particular regular expression.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/4)
+* Check out the solution by Sebastian Aigner in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/09/validating-input-advent-of-code-in-kotlin/) or watch the video:
+
+<video width="560" height="315" href="-kltG4Ztv1s" title="Kotlin Tutorial: Validating and Sanitizing Input. Advent of Code Puzzles"/>
+
+## Day 5. Binary boarding
+
+In this puzzle, you'll use the Kotlin standard library functions to work with the binary representation of numbers,
```

Comment:
We could list them in brackets after (for search engines) with the corresponding links,  if there are not too much functions.